### DISCUSSION TITLE
fprime-util generate missing AC.hpp files : Svc/FileDownlink.hpp

### DISCUSSION THREAD

**Discussion by nikkikaragi** [nasa/fprime#2495] [Q&A] [answered]
When `fprime-util generate`ing a Scheduler component that utilizes a Libcron library, I am getting issues. In Scheduler.hpp, there is the line `#include "Svc/FileDownlink/FileDownlink.hpp"` which then brings an error to the FileDownlink.hpp file and brings about this error. Why isn't this Ac.hpp file being generated? Or is this not even the issue that is occuring? 

<img width="1117" alt="Screenshot 2024-01-21 at 1 07 19 PM" src="https://github.com/nasa/fprime/assets/91200279/ffabe061-8306-46ad-aea3-912f55d758e8">

**Comment by LeStarch** [comment, marked as answer]
If you need a dependency on another component, then you have to add it as a dependency in `CMakeLists.txt`:

`MyProject/Scheduler/CMakeLists.txt:`

```
...
set(MOD_DEPS
    ...
    Svc/FileDownlink
)
...
register_fprime_module()
```
```

**Reply by nikkikaragi** [reply]
Thank you. This makes sense.
